### ISSUE TITLE
butane generates unreadable content for /etc/containers/registries.conf

### ISSUE THREAD

**Issue by freedge** [coreos/butane#417]
using Butane v0.17.0, variant OpenShift version 4.10.0, content is undecodable by OCP 4.11

as https://github.com/openshift/machine-config-operator/pull/2980/commits/918f97bc1c097bd9b6cf045b02b6a8863ccdceac#
never made it in OCP 4.11.

Example of butane config:
```
variant: openshift
version: 4.10.0
metadata:
  name: 99-worker-z-registry
  labels:
    machineconfiguration.openshift.io/role: worker

storage:
  files:
    - path: /etc/containers/registries.conf
      contents:
        inline: |
          foobar
          foobar
          foobar
          foobar
          foobar
          foobar
          foobar
```

error in the machine config daemon:
```
E0120 14:27:13.582590    4838 writer.go:200] Marking Degraded due to: failed decoding TOML content from file /etc/containers/registries.conf: toml: line 1: files cannot contain NULL bytes; probably using
 UTF-16; TOML files must be UTF-8
```

A workaround is to move to version: 4.8.0 (what we did, but using an older version of butane, or OCP 4.12 should work as well).

**Comment by bgilbert** [comment]
openshift/machine-config-operator@918f97bc1c097bd9b6cf045b02b6a8863ccdceac is indeed included in 4.11.  Also, that commit was a refactor; the original commit adding compressed file support was openshift/machine-config-operator@918f97bc1c097bd9b6cf045b02b6a8863ccdceac.

OpenShift generates `registries.conf` from a [custom resource](https://docs.openshift.com/container-platform/4.10/openshift_images/image-configuration.html#images-configuration-file_image-configuration), so one wouldn't normally write out `/etc/containers/registries.conf` from a Butane config.  I've tested with this slightly modified Butane config:

```yaml
variant: openshift
version: 4.10.0
metadata:
  name: 99-worker-z-registry
  labels:
    machineconfiguration.openshift.io/role: worker

storage:
  files:
    - path: /etc/compressed.conf
      contents:
        inline: |
          foobar
          foobar
          foobar
          foobar
          foobar
          foobar
          foobar
```

I applied the MachineConfig generated from this config to a running OCP 4.11.25 cluster and verified that the resulting `/etc/compressed.conf` was uncompressed.

Thus, I'm not able to reproduce a decompression issue.  In addition, if there's an MCO regression, that should be addressed in the MCO rather than in Butane, so I'll go ahead and close this.  If you're still seeing this issue after double-checking your OCP version and config, I'd recommend engaging with OpenShift support.

**Comment by freedge** [comment]
hmm I am running OCP 4.11.25. I opened a case indeed.
Thanks for your inputs.

what butane generates works just fine for any other file, the registries.conf has some special treatment.

My story was that I saw some machineconfig in a git repository and changed the CI to be able to use butane instead, to have more readable files. My issue is with butane deciding to compress this particular file. I will stick with 4.8.0 for the moment and look into not changing this file anymore. Thanks.
